src/output.rs

```diff
@@ -16,6 +17,10 @@ impl<W: Write> Output for WriteOutput<W> {
     fn write(&mut self, seg: &str) -> Result<(), IOError> {
         self.write.write_all(seg.as_bytes())
     }
+
+    fn write_fmt(&mut self, args: std::fmt::Arguments<'_>) -> Result<(), IOError> {
```

Comment:
```suggestion
    fn write_fmt(&mut self, args: std::fmt::Arguments<'_>) -> Result<(), IOError> {
        if let Some(content) = args.as_str() {
            self.write(content)
        } else {
            self.write(&args.to_string())
        }
    }
```

Add a default implementation for compatibility

Reply:
If what I did is not better than your suggestion tel me and I will apply your's had just a quick idea who don't surprise users, but it may be nice to trigger a warning if used as a default impl (not familiar enough with attributes to suggest one)

Thanks for your review time, and sorry for my english I am not a native english writter.

Reply:
(whops commited directly on the `trait` it's late sorry.

Reply:
> Sure, had a long day will I pushed may be better (not sure) added a little bit of documentation to explain why it's better to not stick to the default impl (maybe there is an attribute to warn people using the default impl.

Yes the default implementation is just to keep the code compile if our user has custom `Output`. It should be no bad than their current implementations. For any new `Output`, developer should seek for an optimized implementation.

Reply:
The line:
`            self.write(&args.to_string())` will just print out the format string. This will do no substitution? Tell me if i'am wrong, first time I dig into `std::fmt::Arguments`